plot/plot.app.js

```diff
@@ -1157,16 +1289,128 @@
+                                                                    y:[]
+                                                                }
+                                                                label: //label on which click event took place
+                                                                text:""
+                                                            }
+                                                        }
+                                                    }
+                                                    
+                                                    
+                                                    */
+
+                                                    if(data.hasOwnProperty("points") && data.points[0].data.hasOwnProperty("graphic_link_pattern")){
+                                                        var idx="";
+                                                        switch(data.points[0].data.type){
+                                                            case "pie":
+                                                                idx=data.points[0].data.labels.indexOf(data.points[0].label.toString());
+                                                                break;
+                                                            case "bar":
+                                                                if(data.points[0].data.orientation=="h")
+                                                                    idx=data.points[0].data.y.indexOf(data.points[0].y.toString());
+                                                                else
+                                                                    idx=data.points[0].data.x.indexOf(data.points[0].x.toString());
+                                                                break;
+                                                        }
+                                                        if( data.points[0].data.graphic_link_pattern[idx]!=false && data.points[0].data.graphic_link_pattern[idx]!=undefined){
+                                                            var url=data.points[0].data.graphic_link_pattern[idx];
+                                                            window.open(url,'_blank');
+                                                        }
+                                                    }
+                                                }),
+                                                element[0].on('plotly_legendclick', function(data){
+                                                    //This event occurs on click of legend
+                                                    if(data.hasOwnProperty("data")){
+                                                            if(data.data[0].hasOwnProperty("legend_clickable_links") ){
+                                                                var idx=data.data[0].labels.indexOf(data.label.toString());
```

Comment:
same comment last time I reviewed this PR, can you detail what the "data" looks like for the `legendclick` event?

If it's the same as the object returned in the `plotly_click` event, then state that is the case. I see below that there is also `data.layout`. I didn't see that described above so either redescribe the data object here with the extra value or update the description above to describe what `layout` is. 

Seems like data is the same object for both click events, that means your comment describing what data is should be outside of the `.on` calls.

Reply:
Data for plotly_click and plotly_legendclick are different..